### PR TITLE
doxygen + travis + readme + spi 

### PR DESCRIPTION
**Pull request by hoffmannjan**
doxygen + travis + readme + spi

### CODE REVIEW

File: Adafruit_STMPE610.cpp [hunk 1]
```diff
@@ -1,92 +1,139 @@
...
+#include "WProgram.h"
 #endif
 
 #include <Wire.h>
 #include <SPI.h>
 
 #include "Adafruit_STMPE610.h"
 
-#if defined (SPI_HAS_TRANSACTION)
+#if defined(SPI_HAS_TRANSACTION)
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: remove non-transactions!

File: Adafruit_STMPE610.cpp [hunk 2]
```diff
@@ -1,92 +1,139 @@
...
 
-/**************************************************************************/
-/*! 
-    @brief  Instantiates a new STMPE610 class
-*/
-/**************************************************************************/
-// software SPI
-Adafruit_STMPE610::Adafruit_STMPE610(uint8_t cspin, uint8_t mosipin, uint8_t misopin, uint8_t clkpin) {
+/*!
+ *  @brief  Instantiates a new STMPE610 class
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: using bitbang SPI

File: Adafruit_STMPE610.cpp [hunk 3]
```diff
@@ -1,92 +1,139 @@
...
+                                     uint8_t misopin, uint8_t clkpin) {
   _CS = cspin;
   _MOSI = mosipin;
   _MISO = misopin;
   _CLK = clkpin;
 }
 
-// hardware SPI
+/*!
+ *  @brief  Instantiates a new STMPE610 class
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: using hardware SPI

File: Adafruit_STMPE610.cpp [hunk 4]
```diff
@@ -1,92 +1,139 @@
...
   if (_CS != -1 && _CLK == -1) {
     // hardware SPI
     pinMode(_CS, OUTPUT);
     digitalWrite(_CS, HIGH);
-    
-#if defined (SPI_HAS_TRANSACTION)
-    SPI.begin();
+
+#if defined(SPI_HAS_TRANSACTION)
+    _spi->begin();
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: transactions only

File: Adafruit_STMPE610.cpp [hunk 5]
```diff
@@ -96,259 +143,339 @@ boolean Adafruit_STMPE610::begin(uint8_t i2caddr) {
...
+ *  @brief  Returns true if touched, false otherwise
+ *  @return True if if touched, false otherwise
+ */
+boolean Adafruit_STMPE610::touched() {
   return (readRegister8(STMPE_TSC_CTRL) & 0x80);
 }
 
-boolean Adafruit_STMPE610::bufferEmpty(void) {
+/*!
+ *  @brief  Returns true if touched, false otherwise
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: fix

File: Adafruit_STMPE610.cpp [hunk 6]
```diff
@@ -96,259 +143,339 @@ boolean Adafruit_STMPE610::begin(uint8_t i2caddr) {
...
+    _spi->transfer(x);
 #endif
-  }
-  else
+  } else
     shiftOut(_MOSI, _CLK, MSBFIRST, x);
 }
 
+/*!
+ *  @brief  Reads data from specified register
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: 8 bits of

File: Adafruit_STMPE610.cpp [hunk 7]
```diff
@@ -96,259 +143,339 @@ boolean Adafruit_STMPE610::begin(uint8_t i2caddr) {
...
+    if (_CLK == -1)
+      _spi->endTransaction();
 #endif
   }
 
   return x;
 }
+
+/*!
+ *  @brief  Reads data from specified register
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: 16 bits of

File: Adafruit_STMPE610.cpp [hunk 8]
```diff
@@ -96,259 +143,339 @@ boolean Adafruit_STMPE610::begin(uint8_t i2caddr) {
...
 
-  //Serial.print("$"); Serial.print(reg, HEX); 
-  //Serial.print(": 0x"); Serial.println(x, HEX);
+  // Serial.print("$"); Serial.print(reg, HEX);
+  // Serial.print(": 0x"); Serial.println(x, HEX);
   return x;
 }
 
+/*!
+ *  @brief  Writes data to specified register
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: 8 bits of

**Comment by hoffmannjan** [comment]
all seems to be ok, only I can't pust `gh-pages` branch to satisfy travis because of 403

File: Adafruit_STMPE610.cpp [hunk 9]
```diff
@@ -1,93 +1,119 @@
...
+/*!
+ *  @brief  Instantiates a new STMPE610 using provided SPI
+ *  @param  cspin
+ *          CS pin
+ *  @param  clkpin
+ *          CLK pin
+ *  @param  *theSPI
+ *          spi object
+ */
+Adafruit_STMPE610::Adafruit_STMPE610(uint8_t cspin, uint8_t clkpin,
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: _CLK should be -1

Comment (ladyada) [reply Adafruit_STMPE610.cpp]: (and no clkpin passed in)

File: Adafruit_STMPE610.cpp [hunk 10]
```diff
@@ -1,93 +1,108 @@
...
+Adafruit_STMPE610::Adafruit_STMPE610(TwoWire *theWire) {
+  _CS = _MISO = _MOSI = _CLK = -1;
+  _wire = theWire;
+}
+
+
+/*!
+ *  @brief  Instantiates a new STMPE610 class
+ */
 Adafruit_STMPE610::Adafruit_STMPE610() {
```
Comment (ladyada) [inline Adafruit_STMPE610.cpp]: this is a version of the above function

**Pull request merged**